robot/impl/resource_manager_test.go

```diff
@@ -540,6 +543,162 @@ func TestManagerAdd(t *testing.T) {
 	test.That(t, objectSegmentationService, test.ShouldEqual, injectVisionService)
 }
 
+func TestManagerDuplicateResourceNames(t *testing.T) {
+	fakeModel := resource.DefaultModelFamily.WithModel("fake")
+
+	arm1 := resource.Config{
+		Name:      "arm1",
+		Model:     fakeModel,
+		API:       arm.API,
+		DependsOn: []string{"board1"},
+	}
+	cfg := &config.Config{
+		Components: []resource.Config{arm1},
+	}
+	logger := golog.NewTestLogger(t)
+
+	robotForRemote := &localRobot{
+		manager: newResourceManager(resourceManagerOptions{}, logger),
+	}
+
+	var dummyRemoveOrphanedResourcesCallCount atomic.Uint64
+	dummyRemoveOrphanedResources := func(context.Context, []resource.Name) {
+		dummyRemoveOrphanedResourcesCallCount.Add(1)
+	}
+
+	parentAddr, err := os.MkdirTemp("", "viam-test-*")
+	test.That(t, err, test.ShouldBeNil)
+	mgr := modmanager.NewManager(parentAddr, logger, modmanageroptions.Options{
+		UntrustedEnv:            false,
+		RemoveOrphanedResources: dummyRemoveOrphanedResources,
+	})
+	robotForRemote.manager.moduleManager = mgr
+
+	diff, err := config.DiffConfigs(config.Config{}, *cfg, true)
+	test.That(t, err, test.ShouldBeNil)
+
+	diff.Added.Components = append(diff.Added.Components, arm1)
+
+	err = robotForRemote.manager.updateResources(context.Background(), diff)
+	test.That(t, err, test.ShouldNotBeNil)
+	test.That(t, err.Error(), test.ShouldContainSubstring, "already exists on the robot")
+	test.That(t, err.Error(), test.ShouldContainSubstring, "arm1")
+
+	err = robotForRemote.manager.removeUniqueResource(arm1.ResourceName().String())
+	test.That(t, err, test.ShouldBeNil)
+	diff.Added.Components = make([]resource.Config, 0)
+
+	// check services
+	s1 := resource.Config{
+		Name:  "my-service",
+		Model: fakeModel,
+		API:   arm.API,
+	}
+	cfg.Services = []resource.Config{s1}
+
+	diff.Added.Services = append(diff.Added.Services, s1)
+
+	err = robotForRemote.manager.updateResources(context.Background(), diff)
+	test.That(t, err, test.ShouldBeNil)
+
+	err = robotForRemote.manager.updateResources(context.Background(), diff)
+	test.That(t, err, test.ShouldNotBeNil)
+	test.That(t, err.Error(), test.ShouldContainSubstring, "already exists on the robot")
+	test.That(t, err.Error(), test.ShouldContainSubstring, "my-service")
+
+	// check that removing all services, components + remotes work as expected
```

Comment:
This seems to just be removing the service.

Reply:
yeah sorry will fix comment